src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -167,52 +159,42 @@ private boolean isActiveSession(final String subscriptionId, final String stream
-                        if (Cursor.BEFORE_OLDEST_OFFSET.equals(cursor.getOffset())) {
-                            cursorToProcess = new SubscriptionCursor(cursor.getPartition(), "-1", cursor.getEventType(),
-                                    cursor.getCursorToken());
-                        }
                         final NakadiCursor nakadiCursor = new NakadiCursor(
-                                eventType.getTopic(),
-                                cursorToProcess.getPartition(),
-                                cursorToProcess.getOffset());
+                                cursor.getTimeline(),
+                                cursor.getPartition(),
+                                cursor.getOffset());
                         try {
-                            timelineService.getTopicRepository(eventType).validateCommitCursor(nakadiCursor);
+                            timelineService.getTopicRepository(cursor.getTimeline()).validateCommitCursor(nakadiCursor);
                             return nakadiCursor;
                         } catch (final InvalidCursorException e) {
                             throw new NakadiRuntimeException(e);
                         }
                     })
                     .collect(Collectors.toList());
 
-            LOG.debug("[COMMIT_CURSORS] finished validation of {} cursor(s) for partition {}", cursors.size(),
-                    eventTypePartition);
+            LOG.debug("[COMMIT_CURSORS] finished validation of {} cursor(s) for partition {} {}", cursors.size(),
+                    cursors.get(0).getTimeline().getEventType(), cursors.get(0).getTopic());
 
-            return commitPartitionCursors(subscriptionId, eventType, nakadiCursors, eventTypePartition);
+            return commitPartitionCursors(subscriptionId, nakadiCursors);
 
         } catch (final NakadiRuntimeException e) {
             throw (InvalidCursorException) e.getException();
         }
     }
 
-    private List<Boolean> commitPartitionCursors(final String subscriptionId,
-                                                 final EventType eventType,
-                                                 final List<NakadiCursor> cursors,
-                                                 final EventTypePartition etPartition)
+    private List<Boolean> commitPartitionCursors(final String subscriptionId, final List<NakadiCursor> cursors)
             throws ServiceUnavailableException {
-        final String topic = eventType.getTopic();
-        final String offsetPath = format(PATH_ZK_OFFSET, subscriptionId, topic, etPartition.getPartition());
-        final TopicRepository topicRepository = timelineService.getTopicRepository(eventType);
+        final NakadiCursor first = cursors.get(0);
+        final String offsetPath = format(
+                PATH_ZK_OFFSET, subscriptionId, first.getTopic(), first.getPartition());
+        final TopicRepository topicRepository = timelineService.getTopicRepository(first.getTimeline());
```

Comment:
@v-stepanov This PR is part of LOLA timelines part. When one will get to Hila part, then we will support it